NetworkPkg/UefiPxeBcDxe/PxeBcBoot.c

```diff
@@ -1268,5 +1268,12 @@ PxeBcLoadBootFile (
     AsciiPrint ("\n  PXE-E99: Unexpected network error.\n");
   }
 
+  REPORT_STATUS_CODE_WITH_EXTENDED_DATA (
+    EFI_ERROR_CODE,
+    (EFI_STATUS_CODE_VALUE)(EFI_IO_BUS_IP_NETWORK | EFI_OEM_SPECIFIC | ((EFI_STATUS_CODE_VALUE)(Status & 0x1F))),

```

Comment:
Can you explain the usage of EFI_OEM_SPECIFIC here? Since we're reporting all errors here, maybe instead of OEM_SPECIFIC we can use EFI_IOB_EC_NON_SPECIFIC?

Reply:
@SaloniKasbekar, since the status code we reported here is actually the `((EFI_STATUS_CODE_VALUE)(Status & 0x1F))`, which is not a standard definition in PI SPEC, so we use `EFI_OEM_SPECIFIC` to indicate it is non-standard PI definition here, we can add one more `| EFI_IOB_EC_NON_SPECIFIC` here but it will have no impact to the result, and the `EFI_OEM_SPECIFIC` should still remain here for the non-standard status code.

Please let us know if you want to add `| EFI_IOB_EC_NON_SPECIFIC` or any other suggestion to handle the `((EFI_STATUS_CODE_VALUE)(Status & 0x1F))`.

Reply:
Thats fair. No other concerns with the changes. Will approve.

@mdkinney Can you help review/push?